controller/Delivery.php

```diff
@@ -0,0 +1,432 @@
+    }
+
+    /**
+     * List available test takers to assign to a delivery
+     *
+     * @throws \Exception
+     * @throws \common_Exception
+     * @throws \oat\oatbox\service\ServiceNotFoundException
+     */
+    public function testTakers() {
+
+        $delivery = $this->getCurrentDelivery();
+        $testCenter = $this->getCurrentTestCenter();
+        
+        try {
+
+            $deliveryService = $this->getServiceManager()->get('taoProctoring/delivery');
+            $currentUser = \common_session_SessionManager::getSession()->getUser();
+
+            $requestOptions = $this->getRequestOptions();
+            $users = $deliveryService->getAvailableTestTakers($currentUser, $delivery->getUri(), $requestOptions);
+            $testTakers = $this->getTestTakersPage($users, $requestOptions);
+
+            $this->composeView(
+                'assign-test-takers',
+                array(
+                    'id' => $delivery->getUri(),
+                    'title' =>  __('Assign test takers to %s', $delivery->getLabel()),
+                    'testsite' => $testCenter->getUri(),
+                    'set' => $testTakers //change it to list for consistency
+                ),array(
+                    Breadcrumbs::testCenters(),
+                    Breadcrumbs::testCenter($testCenter, $this->getTestCenters()),
+                    Breadcrumbs::deliveries($testCenter,
+                        array(
+                            Breadcrumbs::diagnostics($testCenter),
+                            Breadcrumbs::reporting($testCenter)
+                        )
+                    ),
+                    Breadcrumbs::deliveryMonitoring($testCenter, $delivery, $this->getDeliveries()),
+                    Breadcrumbs::deliveryTestTaker($testCenter, $delivery)
+                )
+            );
+
+        } catch (ServiceNotFoundException $e) {
+            \common_Logger::w('No delivery service defined for proctoring');
+            $this->returnError('Proctoring interface not available');
+        }
+
+    }
+
+    /**
+     * Gets the value of a string property from a user
+     * @param User $user
+     * @param string $property
+     * @return mixed|string
+     */
+    private function getUserStringProp($user, $property) {
+        $value = $user->getPropertyValues($property);
+        return empty($value) ? '' : current($value);
+    }
+
+    /**
+     * Get a list of test takers usable in a table page
+     * @param User[] $users
+     * @param array $options
+     * @return array
+     */
+    private function getTestTakersPage($users, $options) {
+        $amount = count($users);
+        $rows = max(1, abs(ceil(isset($options['rows']) ? $options['rows'] : 25)));
+        $total = ceil($amount / $rows);
+        $page = max(1, floor(min(isset($options['page']) ? $options['page'] : 1, $total)));
+        $start = ($page - 1) * $rows;
+        $list = array();
+
+        $users = array_slice($users, ($page - 1) * $rows, $rows);
+
```

Comment:
Woops, this method has been refactored to make use of the paginate method, maybe you miss it